var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -251,7 +251,8 @@ class Cp2k(MakefilePackage, CudaPackage, CMakePackage, ROCmPackage):
     depends_on("wannier90", when="@3.0+mpi")
 
     with when("build_system=cmake"):
-        depends_on("dbcsr")
+        depends_on("dbcsr@2.6.0", when="@2023.2:")
+        depends_on("dbcsr", when="@:2023.1")
```

Comment:
Does CP2K work with _any_ version of dbcsr until 2023.1?

If that's the case, I would keep the generic `depends_on` and only include a constraint for `2023.2`. I.e.
```suggestion
        depends_on("dbcsr")
        depends_on("dbcsr@2.6.0", when="@2023.2:")
```
Additionally, if there's no newer version of dbcsr yet, I would not require an exact version of it, but put a lower bound:
```suggestion
        depends_on("dbcsr")
        depends_on("dbcsr@2.6:", when="@2023.2:")
```

Reply:
cp2k depends on specific version of dbcsr (because dbcsr changed API the past years) that's why we have to specialize.